src/persist-client/src/internal/state_versions.rs

```diff
@@ -831,17 +854,51 @@ impl<T: Timestamp + Lattice + Codec64> StateVersionsIter<T> {
             key_codec,
             val_codec,
             diff_codec,
+            #[cfg(debug_assertions)]
+            validator: ReferencedBlobValidator::default(),
         }
     }
 
-    pub fn next(&mut self) -> Option<&State<T>> {
+    // TODO: This closure feels a bit over-indexed on the callers that use it,
+    // but also it feels too early to figure out what a nice generalization is.
```

Comment:
👍 I'll explore these

Reply:
did the first one!

the second felt ended up feeling like a bit of a footgun to me, because they both would have advanced through versions. if we could totally separate iterators for diffs vs states, I think that would have worked nicely, but in practice the ones that want to look at diffs also want to look at the end state, so there would have been too much overlap